src/ParallelRun.jl

```diff
@@ -153,61 +304,79 @@ function infer_init(rcfs::Vector{RunCamcolField},
-        append!(source_cat_idxs, these_sources_idxs)
+    # check if any non-default values are passed for functionality that has
+    # been removed.
+    if !primary_initialization
+        error("primary_initialization no longer supported.")
+    end
+    if !(objid == "")
+        error("objid no longer supported")
     end
-    timing.read_photoobj += toq()
 
-    # Get indices of entries in the RA/Dec range of interest.
-    entry_in_range = entry->((box.ramin < entry.pos[1] < box.ramax) &&
-                             (box.decmin < entry.pos[2] < box.decmax))
-    target_sources = find(entry_in_range, catalog)
+    # Initialize variables to empty vectors in case try block fails
+    catalog = CatalogEntry[]
+    source_rcfs = RunCamcolField[]
+    source_cat_idxs = Int16[]
+    source_radii = Float64[]
+    target_sources = Int[]
+    images = Image[]
+
+    try
+        # Read in images for all RCFs
+        tic()
+        raw_images = SDSSIO.load_raw_images(strategy, rcfs)
+        timing.read_img += toq()
 
-    Log.info("$(Time(now())): $(length(catalog)) primary sources, ",
-             "$(length(target_sources)) target sources in $(box.ramin), ",
-             "$(box.ramax), $(box.decmin), $(box.decmax)")
+        # detect sources on all raw images (before background added back)
+        catalog, source_rcfs, source_cat_idxs, source_radii =
+            detect_sources(raw_images)
 
-    # Filter any object not specified, if an objid is specified
-    if objid != ""
-        target_sources = filter(ts->(catalog[ts].objid == objid), target_sources)
-        Log.info("$(length(target_sources)) target light sources after objid cut")
-    end
+        # Get indices of entries in the RA/Dec range of interest.
+        # (Some images can have regions that are outside the box, so not
+        # all sources are necessarily in the box.)
+        entry_in_range = entry->((box.ramin < entry.pos[1] < box.ramax) &&
+                                 (box.decmin < entry.pos[2] < box.decmax))
+        target_sources = find(entry_in_range, catalog)
+
+        Log.info("$(Time(now())): $(length(catalog)) primary sources, ",
+                 "$(length(target_sources)) target sources in $(box.ramin), ",
+                 "$(box.ramax), $(box.decmin), $(box.decmax)")
 
-    # Load images and neighbor map for target sources
-    if length(target_sources) > 0
-        # Read in images for all (run, camcol, field).
+        # convert raw images to images
         try
-            tic()
-            images = SDSSIO.load_field_images(strategy, rcfs)
-            timing.read_img += toq()
-        catch ex
-            Log.exception(ex)
-            empty!(target_sources)
+            images = [convert(Image, raw_image) for raw_image in raw_images]
+        catch exc
+            Log.exception(exc)
+            rethrow()
         end
 
-        tic()
-        neighbor_map = Infer.find_neighbors(target_sources, catalog, images)
```

Comment:
Still used in the accuracy benchmarks at the moment, but can be deleted once that is updated.